### ISSUE TITLE
Remove legacy code related to w3c-common profile

### ISSUE THREAD

**Issue by sidvishnoi** [w3c/respec#2889]
- [ ] Remove code due to legacy support
- [ ] Fix highlightjs dynamic language loading - uses accessing worker with `require`
- [ ]  Fix tools/respecDocWriter.js - uses exporter with `require`
- [ ] Remove `require` support - do not `expose` anymore - BREAKING CHANGE

**Comment by sidvishnoi** [comment]
Before we start deprecation warning, we should add a migration guide.
As we plan to remove `require(["some-respec-internal-thing"])`, we should figure out a strategy for extensions also (See https://github.com/w3c/respec/issues/1976), otherwise migration is like "check back later".

**Comment by JohnRiv** [comment]
I'm not sure if https://github.com/w3c/respec/wiki/respec-w3c-common-migration-guide was ever created, but that page does not exist anymore

**Comment by sidvishnoi** [comment]
Damn bots deleted page. Thanks for reporting. It's available again.
![image](https://user-images.githubusercontent.com/8426945/94987904-49b11d00-0587-11eb-99d6-34aef04a91a3.png)

**Comment by carmacleod** [comment]
The migration guide is gone again, fyi.

**Comment by sidvishnoi** [comment]
Thanks for reporting @carmacleod. Added it back again, and restricted wiki edit access for some time.

**Comment by marcoscaceres** [comment]
The w3c-common deprecation tracked here is effectively done: the profile is frozen, the deprecation warning is in place, the migration guide exists, jQuery is removed, and legacy code was cleaned up (6 of 9 items checked).

The remaining unchecked items are tracked separately: extensions API (#1976), hljs dynamic loading (#4934), and removing `require` support (would need its own issue as a breaking change). Closing this umbrella tracker. Thanks @sidvishnoi for driving this.
